config/validator/service/proposer_settings.go

```diff
@@ -0,0 +1,203 @@
+package validator_service_config
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	fieldparams "github.com/prysmaticlabs/prysm/v4/config/fieldparams"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/validator"
+	"github.com/prysmaticlabs/prysm/v4/encoding/bytesutil"
+	validatorpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1/validator-client"
+)
+
+// ToSettings converts struct to ProposerSettings
+func ToSettings(ps *validatorpb.ProposerSettingsPayload) (*ProposerSettings, error) {
+	settings := &ProposerSettings{}
+	if ps.ProposerConfig != nil {
+		settings.ProposeConfig = make(map[[fieldparams.BLSPubkeyLength]byte]*ProposerOption)
+		for key, optionPayload := range ps.ProposerConfig {
+			if optionPayload.FeeRecipient == "" {
+				continue
+			}
+			b, err := hexutil.Decode(key)
+			if err != nil {
+				return nil, err
```

Comment:
We should add info about what the bad key was that we were trying to decode here

Reply:
errorf perhaps and %s

Reply:
fixed